src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -43,28 +43,37 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
         {
             base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
 
-            CSharpCompilation compilation = this.DeclaringCompilation;
+            AddSynthesizedFieldAttributes(ref attributes, this, this.SuppressDynamicAttribute);
+        }
+
+        internal static void AddSynthesizedFieldAttributes(
```

Comment:
>AddSynthesizedFieldAttributes [](start = 29, length = 29)

It feels like, if we move SourceFieldSymbol.AddSynthesizedAttributes to the new base, we wouldn't have to make any changes in this file.